### PR TITLE
#5656 Add makeAddressActual endpoint

### LINKED ISSUE #5656: [UBS-User.Edit Profile] User cannot save the default address when he/she selects it on the editable 'User data' page
**Environment:** Windows 10 Pro v.21H2, Google Chrome 111.0.5563.147 (64-bit).
**Reproducible:** always.
**Build found:** 07/04/23

**Preconditions**

1. User is logged in as user.
2. User data tab is opened.
3. More than 1 address is added.

**Steps to reproduce**
1. Click on the 'Edit Data' button.
2. Click on the unchecked 'Make the address by default' checkbox for any address except the first one.
e.g. Address 3.
4. Click on the 'Save changes' button.
5. Click on the 'Edit Data' button.
6. Pay attention to the 'Make it the default address' checkbox for this address.
7. Click on the 'Orders' tab in the navigation menu.
8. Click on the 'New order' button''.
9. Go to the 'Personal data' tab.
10. Pay attention to the default address in the 'Garbage collection address' section.

**Actual result**
'Make the address by default' checkbox is unchecked for the selected address on the editable 'User data' page. The default address isn't changed in the Order form.

https://user-images.githubusercontent.com/119864646/230674959-b1ba0796-46e1-4d1b-9ca0-2fc997c8825b.mp4

**Expected result**
'Make the address by default' checkbox should be checked for the selected address on the editable 'User data' page. The default address in the Order form should be matched with the address by default in the editable 'User data' page.

**User story and test case links**
User story #3811 
[Test case](https://jira.softserve.academy/browse/GC-2828)

**Labels to be added**
"Bug", Priority ("pri: High"), Severity ("severity: Major"), Type "Functional").

### PR DESCRIPTION
**Pull request by sergoliarnik**
dev
## JIRA

https://github.com/ita-social-projects/GreenCity/issues/5656

## Code reviewers

- [ ] @github_username

## Summary of issue

Add makeAddressActual endpoint

## Summary of change

Add makeAddressActual controller and service methods, add new tests

## Testing approach

ToDo

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 1]
```diff
@@ -1765,4 +1765,31 @@ public List<LocationSummaryDto> getLocationSummary() {
...
+     */
+    @Override
+    @Transactional
+    public OrderWithAddressesResponseDto makeAddressActual(Long addressId, String uuid) {
+        Address currentAddress = addressRepo.findById(addressId).orElseThrow(
+            () -> new NotFoundException(NOT_FOUND_ADDRESS_ID_FOR_CURRENT_USER + addressId));
+
+        User user = userRepository.findByUuid(uuid);
+
+        if (!currentAddress.getUser().equals(user)) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Can we check if `currentAddress.getUser().getUuid.equals(uuid)`?
In this case, we do not need to call the database for the user

File: service-api/src/main/java/greencity/service/ubs/UBSClientService.java [hunk 2]
```diff
@@ -366,4 +366,13 @@ public interface UBSClientService {
      * @author Max Nazaruk
      */
     List<LocationSummaryDto> getLocationSummary();
+
+    /**
+     * Method makes address actual (default) for current {@link User}.
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/service/ubs/UBSClientService.java]: This method makes address with `addresseId` actual for the user with `uuid` not do r current user

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 3]
```diff
@@ -1765,4 +1765,31 @@ public List<LocationSummaryDto> getLocationSummary() {
...
+            throw new AccessDeniedException(CANNOT_DELETE_ADDRESS);
+        }
+
+        if (AddressStatus.DELETED.equals(currentAddress.getAddressStatus())) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: for enums, we can use `==`
`AddressStatus.DELETED == currentAddress.getAddressStatus()`

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 4]
```diff
@@ -1765,4 +1765,31 @@ public List<LocationSummaryDto> getLocationSummary() {
             .map(location -> modelMapper.map(location, LocationSummaryDto.class))
             .collect(toList());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @Transactional
+    public OrderWithAddressesResponseDto makeAddressActual(Long addressId, String uuid) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Why does `makeAddressActual` return `OrderDTO`?

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 5]
```diff
@@ -1765,4 +1765,31 @@ public List<LocationSummaryDto> getLocationSummary() {
...
+        if (AddressStatus.DELETED.equals(currentAddress.getAddressStatus())) {
+            throw new BadRequestException(CANNOT_MAKE_ACTUAL_DELETED_ADDRESS);
+        }
+
+        List<Address> addresses = addressRepo.findAllNonDeletedAddressesByUserId(user.getId());
+
+        addresses.forEach(address -> address.setActual(false));
+        currentAddress.setActual(true);
+
+        return findAllAddressesForCurrentOrder(uuid);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Why do you use `userUuid` as `orderId`?

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 6]
```diff
@@ -1765,4 +1765,31 @@ public List<LocationSummaryDto> getLocationSummary() {
...
+
+        if (!currentAddress.getUser().equals(user)) {
+            throw new AccessDeniedException(CANNOT_DELETE_ADDRESS);
+        }
+
+        if (AddressStatus.DELETED.equals(currentAddress.getAddressStatus())) {
+            throw new BadRequestException(CANNOT_MAKE_ACTUAL_DELETED_ADDRESS);
+        }
+
+        List<Address> addresses = addressRepo.findAllNonDeletedAddressesByUserId(user.getId());
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: Can the user have more then obe actual address?
Why cant we get only the actual address form DB?

File: core/src/main/java/greencity/controller/OrderController.java [hunk 7]
```diff
@@ -281,6 +282,29 @@ public ResponseEntity<OrderWithAddressesResponseDto> deleteOrderAddress(
...
+    /**
+     * Controller make address actual (default).
+     *
+     * @param addressId {@link Long}.
+     * @param uuid      {@link UserVO} id.
+     * @return {@link HttpStatus} - http status.
+     */
+    @ApiOperation(value = "Make address actual (default)")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK, response = OrderWithAddressesResponseDto.class),
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/OrderController.java]: this endpoint returns `AddressDto`

File: core/src/main/java/greencity/controller/OrderController.java [hunk 8]
```diff
@@ -281,6 +282,29 @@ public ResponseEntity<OrderWithAddressesResponseDto> deleteOrderAddress(
             .body(ubsClientService.deleteCurrentAddressForOrder(id, uuid));
     }
 
+    /**
+     * Controller make address actual (default).
+     *
+     * @param addressId {@link Long}.
+     * @param uuid      {@link UserVO} id.
+     * @return {@link HttpStatus} - http status.
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/OrderController.java]: this method returns `ResponseEntity<AddressDto> `

**Review by RomaMocherniuk — APPROVED** [review]

**Review by Maksym-Lenets — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Pull request merged**